### DISCUSSION TITLE
Disable repeatable relationship automatic linked models creation?

### DISCUSSION THREAD

**Discussion by julienmru** [Laravel-Backpack/community-forum#741] [Q&A (Help)]
I updated an old project on Backpack 4.1 where I had custom code to create children items on a `repeatable` field. 

**Here's my setup:**

I have `Event` model & crud controller, on which I set a repeatable for a field `dates`:

``` 
       CRUD::addField([
            'name'  => 'dates',
            'label' => 'Dates',
            'type'  => 'repeatable',
            'fields' => [
                  ...
            ]
       );
```

`Date` is also a model, with the relation defined in `Event`:

```
class Event extends Model implements HasMedia
{
    ....
    public function dates()
    {
        return $this->hasMany(Date::class)->orderBy('starts_at', 'asc');
    }
    ....
}
```

Let me know if that's not clear : )

**Here's my issue:**

Backpack has evolved (well done!) and automagically creates the items in a relation when using a `repeatable` field (from the `Event` controller to the `Date` model in my case) but it's not working as I'd expect it to work because it deletes all linked `Date` and creates new ones, thus changing my IDs.

_Long story short_, **is there a way to disable the creation on linked models** in Backpack? I'd like to keep some logic at my end.

**Comment by pxpm** [comment]
Hey @julienmru I think `'entity => false'` is what you are looking for, and for "complete" control you can just overwrite backpack `create` and `update` functions from `CreateOperation` and `UpdateOperation` respectively. 

AFAIK, HasMany shouldn't be deleting and recreating, as it runs `updateOrCreate()` https://github.com/Laravel-Backpack/CRUD/blob/fa64c2eba69ddbc85eeb698f283ecbe0fd925669/src/app/Library/CrudPanel/Traits/Create.php#L312

Maybe something is not properly setup, or you are using some previous buggy version? You can check that with `php artisan backpack:version`. 

If you want to keep exploring this and/or entity false does not work for you have a look at the [HasMany relation docs](https://backpackforlaravel.com/docs/6.x/crud-how-to#hasmany-1-n-relationship) 

Let me know if that helps.

Cheers

**Comment by jcastroa87** [comment]
Hi @julienmru did you have a chance to check about @pxpm answer?

Please, let us know if you need anything else. Meanwhile, i will close the issue, but please feel free to re-open or create a new one if needed.
